scipy/signal/_spectral_py.py

```diff
@@ -171,6 +172,10 @@ def periodogram(x, fs=1.0, window='boxcar', nfft=None, detrend='constant',
         required parameters. If `window` is array_like it will be used
         directly as the window and its length must be nperseg. Defaults
         to 'boxcar'.
+    nperseg : int, optional
```

Comment:
It does not make sense to me to add `nperseg` as an argument. Note that in `welch`, the `nperseg` parameter does make sense because Welch's method is about dividing into segments (see first paragraph of its docstring). For `periodogram` however, there is no such thing - it using `welch` under the hood is an implementation detail.

I don't think this is needed to resolve the problem, right?